extensions/default/src/MergeDataSource/index.js

```diff
@@ -0,0 +1,195 @@
+import { DicomMetadataStore, IWebApiDataSource } from '@ohif/core';
+import { get } from 'lodash';
+
+/**
+ * Returns deduplicated data by comparing the incomingData with the mergedData based on the mergeKey.
+ * @param {Object} options - The options object.
+ * @param {Array} options.mergedData - The global data array.
+ * @param {Array} options.incomingData - The incoming data array.
+ * @param {string} options.mergeKey - The merge key.
+ * @returns {Array} The deduplicated data array.
+ */
+const getDedupedData = ({ mergedData, incomingData, mergeKey }) => {
+  const keys = mergedData.map(r => get(r, mergeKey));
+  return incomingData.filter(r => !keys.includes(get(r, mergeKey)));
```

Comment:
It is quite a bit more efficient to do
mergedData.has(get(r,mergeKey)) instead of keys.includes, which performs a linear search.  On 2000 item series, that is 2,000,000 comparisons - definitely something to avoid if possible.

Reply:
What you end up doing is const key = get(r,mergeKey);  mergedData.set(key, r)  You can then also add it to a return list, OR you can choose which version to include based on other conditions (future work)

Reply:
I'm not sure about has, this is not part of array prototype, can you sketch some code sample?